client/verta/verta/endpoint/_endpoint.py

```diff
@@ -186,22 +253,35 @@ def update(self, model_reference, strategy=None, wait=False, resources=None,
             Autoscaling condition for the updated endpoint.
         env_vars : dict of str to str, optional
             Environment variables.
+        kafka_settings : :class:`verta.endpoint.KafkaSettings` or False, optional
+            Kafka settings. If ``False``, clears this endpoint's existing Kafka
+            settings.
 
         Returns
         -------
         status : dict of str to {None, bool, float, int, str, list, dict}
 
         """
         if not isinstance(model_reference, (RegisteredModelVersion, ExperimentRun)):
-            raise TypeError("`model_reference` must be an ExperimentRun or RegisteredModelVersion")
+            raise TypeError(
+                "`model_reference` must be an ExperimentRun or RegisteredModelVersion"
+            )
 
         if not strategy:
             strategy = DirectUpdateStrategy()
 
-        update_body = self._create_update_body(strategy, resources, autoscaling, env_vars)
+        update_body = self._create_update_body(
+            strategy, resources, autoscaling, env_vars
+        )
+
+        # update Kafka settings
+        if kafka_settings is False:
+            self._del_kafka_settings(self._get_or_create_stage())
+        elif kafka_settings:
+            self._set_kafka_settings(self._get_or_create_stage(), kafka_settings)
```

Comment:
Here is where Kafka settings are modified in `endpoint.update()`.

This notably makes `endpoint.update()` less atomic, because updating the Kafka settings of the stage is separate from creating a build.